File: scripts/generate_synthetic_data.js
const fs = require('fs');

function generateData() {
    const data = {
        metadata: {
            subject_id: "SUB-2024-001",
            age_months: 36,
            session_duration_seconds: 10,
            frame_rate: 10
        },
        timeline: [],
        summary_stats: {}
    };

    const totalFrames = 100;
    let caregiverGazeCount = 0;

    for (let i = 0; i < totalFrames; i++) {
        const timestamp = parseFloat((i * 0.1).toFixed(1));

        // Gaze Generation (85% Non-Caregiver)
        const rand = Math.random();
        let gaze;
        if (rand < 0.45) gaze = "Stimulus";
        else if (rand < 0.85) gaze = "Void";
        else gaze = "Caregiver";

        if (gaze === "Caregiver") caregiverGazeCount++;

        // Hand Flapping (Frames 40-70)
        const handFlapping = (i >= 40 && i <= 70);

        // Body Orientation (100-260 degrees)
        const bodyAngle = Math.floor(Math.random() * (260 - 100 + 1)) + 100;

        data.timeline.push({
            frame_id: i,
            timestamp: timestamp,
            gaze_target: gaze,
            hand_flapping_detected: handFlapping,
            body_orientation_angle: bodyAngle
        });
    }

    // Summary Stats
    const jointAttentionRatio = caregiverGazeCount / totalFrames;

    let riskScore = 0;
    if (jointAttentionRatio < 0.2) riskScore += 40;
    if (jointAttentionRatio < 0.1) riskScore += 20;
    if (data.timeline.some(f => f.hand_flapping_detected)) riskScore += 30;
    riskScore += Math.floor(Math.random() * 11);

    data.summary_stats = {
        joint_attention_ratio: jointAttentionRatio,
        risk_score: Math.min(riskScore, 100)
    };

    return data;
}

const dataset = generateData();
fs.writeFileSync('synthetic_screening_data.json', JSON.stringify(dataset, null, 2));
console.log('Dataset generated: synthetic_screening_data.json');

File: scripts/generate_synthetic_data.py
import json
import random

def generate_data():
    data = {
        "metadata": {
            "subject_id": "SUB-2024-001",
            "age_months": 36,
            "session_duration_seconds": 10,
            "frame_rate": 10
        },
        "timeline": [],
        "summary_stats": {}
    }

    # Simulation parameters
    total_frames = 100
    
    # 85% Stimulus/Void (Low Joint Attention), 15% Caregiver
    gaze_options = ["Stimulus", "Void", "Caregiver"]
    gaze_weights = [0.45, 0.40, 0.15] 

    caregiver_gaze_count = 0

    for i in range(total_frames):
        timestamp = round(i * 0.1, 1)
        
        # Gaze Generation
        gaze = random.choices(gaze_options, weights=gaze_weights, k=1)[0]
        if gaze == "Caregiver":
            caregiver_gaze_count += 1
            
        # Hand Flapping (Frames 40-70)
        hand_flapping = 40 <= i <= 70
        
        # Body Orientation (Facing away: 90-270 degrees)
        # 0 is facing caregiver. 180 is back to caregiver.
        # Simulating mostly facing away
        body_angle = random.randint(100, 260)

        frame = {
            "frame_id": i,
            "timestamp": timestamp,
            "gaze_target": gaze,
            "hand_flapping_detected": hand_flapping,
            "body_orientation_angle": body_angle
        }
        data["timeline"].append(frame)

    # Summary Stats
    joint_attention_ratio = caregiver_gaze_count / total_frames
    
    # Mock Risk Score Calculation
    # Low joint attention (< 0.2) increases risk
    # Hand flapping increases risk
    risk_score = 0
    if joint_attention_ratio < 0.2:
        risk_score += 40
    if joint_attention_ratio < 0.1:
        risk_score += 20
    
    # Check for sustained hand flapping
    if any(f["hand_flapping_detected"] for f in data["timeline"]):
        risk_score += 30
        
    # Baseline noise
    risk_score += random.randint(0, 10)
    
    data["summary_stats"] = {
        "joint_attention_ratio": joint_attention_ratio,
        "risk_score": min(risk_score, 100)
    }

    return data

if __name__ == "__main__":
    dataset = generate_data()
    with open("synthetic_screening_data.json", "w") as f:
        json.dump(dataset, f, indent=2)
    print("Dataset generated: synthetic_screening_data.json")
